### PR TITLE
Exclude requirements.in file for clang format

### PR DESCRIPTION
**Pull request by pfultz2**

### CODE REVIEW

File: tools/format.py [hunk 1]
```diff
@@ -21,24 +21,34 @@
...
     return subprocess.run(cmd,
                           capture_output=True,
-                          shell=True,
+                          shell=isinstance(cmd, str),
                           check=True,
                           **kwargs).stdout.decode('utf-8').strip()
 
 
+def is_excluded(f):
+    base = os.path.basename(f)
```
Comment (TedThemistokleous) [inline tools/format.py:48]: Pyflakes is complaining about this

File: tools/format.py [hunk 2]
```diff
@@ -21,24 +21,34 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
```
Comment (TedThemistokleous) [inline tools/format.py:21]: Update license

File: tools/format.py [hunk 3]
```diff
@@ -21,24 +21,34 @@
...
 # THE SOFTWARE.
 #####################################################################################
-import os, shutil, argparse, subprocess
+import os, shlex, shutil, argparse, subprocess
 
 CLANG_FORMAT_PATH = '/opt/rocm/llvm/bin'
 
+EXCLUDE_FILES = ['requirements.in']
```
Comment (causten) [inline tools/format.py:28]: So this doesn't need a path?  The only requirements.in I see in our code base is [docs/sphinx/requirements.in](https://github.com/ROCm/AMDMIGraphX/blob/a2752d8e9e7e0c959a0506988938ec125d048f3e/docs/sphinx/requirements.in)

Comment (pfultz2) [reply tools/format.py:28]: > So this doesn't need a path?

Nope, it checks the basename.

Comment (pfultz2) [reply tools/format.py:28]: I see, there is a mistake there.

Comment (pfultz2) [reply tools/format.py:28]: Fixed.

**Review by CharlieL7 — APPROVED** [review]

**Review by umangyadav — APPROVED** [review]

File: tools/format.py [hunk 4]
```diff
@@ -21,24 +21,34 @@
...
+
 
 def run(cmd, **kwargs):
-    print(cmd)
-    subprocess.run(cmd, shell=True, check=True, **kwargs)
+    if isinstance(cmd, str):
+        print(cmd)
+    else:
+        print(shlex.join(cmd))
```
Comment (umangyadav) [inline tools/format.py:35]: `" ".join(cmd)`
simply doing this doesn't work ?

**Pull request merged**